### DISCUSSION TITLE
Venmo debit card transactions

### DISCUSSION THREAD

**Discussion by drewsmith9** [mmohades/Venmo#55] [General]
I have a use case I have been trying to resolve. I want to reconcile my Venmo debit card transactions to corresponding reloads initiated by Venmo from my bank account. I see that you all are only using TransactionType=PAYMENT now, so I hope soon that you may implement a feature to solve this.

Right now I have a way of using venmo_statements.csv but Selenium can't get around 2FA. Any suggestions?

Anyways, this library is handy!

**Comment by mmohades** [comment]
Yeah I need to add support for other transaction types. Sorry I'm not following, what are you trying to achieve?

**Reply by drewsmith9** [reply]
Venmo provides a debit card that will pull from your Venmo balance to cover transactions. If there are not enough funds in the Venmo wallet to cover, it will pull from the connected bank account in $10 increments until it has enough funds to cover the transaction. I would like to tie the initial transaction to the reload Venmo transaction together in order to know why those funds were pulled. I use Mint for personal finance and will ultimately categorize the transactions later on. My goal is to run code weekly to reconcile these scenarios.

**Reply by mmohades** [reply]
gotchu. You can get all your transactions as a JSON by just making a GET request yourself. Use this code for example to get all your transactions:
```python
from venmo_api import ApiClient, Client

access_token = 'token'
api_client = ApiClient(access_token=access_token)
client = Client(access_token)


resource_path = f'/stories/target-or-actor/{client.my_profile().id}'
response = api_client.call_api(resource_path=resource_path, method='GET')
```
There is no processing happening on the response, so you're basically getting the raw response with all the transactions.

**Reply by drewsmith9** [reply]
does the default to returning the past 21 days of transactions?

**Reply by mmohades** [reply]
The recent 50 transactions. You can get 50 at a time. Refer to here: https://github.com/mmohades/VenmoApiDocumentation#users-transactions-list
Send before_id and limit as request parameters.
